add case study with neurolytics (#15)
localstack/localstack-blog.github.io: content/2024-01-12-localstack-neurolyticsai/index.md

      - S3
      - Lambda
      - Elastic Container Service (ECS)
      - CloudWatch Logs
      - EventBridge
  - key: Integrations
    value:
      - AWS CLI
      - AWS SDK
      - Docker SDK
---

<div class="quote-container mt-4">

  > _“LocalStack has made cloud development easier to test and debug, significantly reducing our development and testing loops by approximately 30%. This enables faster iterations and quicker feature releases and translates to substantial financial savings.”_
  <div class="quote-author">
    <p><a href="https://www.linkedin.com/in/yashar-hosseinpour/">Yashar Hosseinpour</a>,</p>
    <p>Lead Developer at <a href="https://neurolytics.ai/">Neurolytics</a></p>
  </div>
</div>

<div class="lead-content">
  <p>Neurolytics, a startup in the field of behavioral analysis, has developed a platform for enhancing the recruitment process through AI-driven video analysis. Their solution offers a transformative approach for companies and job candidates, ensuring a match between their profiles and the company culture. This is crucial for hiring individuals who are not only qualified but are also likely to thrive and stay long-term within the organization.</p>

>   <p>Hosted on Amazon Web Services (AWS), Neurolytics' infrastructure leverages the robustness and scalability of the platform. However, the challenge of efficient and cost-efficient development in such a dynamic environment led them to LocalStack. This move has enabled the Neurolytics team to significantly streamline their development and testing cycles significantly, thereby enhancing their agility and innovation capabilities.</p>

Reviewer: ```suggestion
  <p>Hosted on Amazon Web Services (AWS), Neurolytics' infrastructure leverages the robustness and scalability of the platform. However, the challenge of time- and cost-efficient development in such a dynamic environment led them to LocalStack. This move has enabled the Neurolytics team to significantly streamline their development and testing cycle, thereby enhancing their agility and innovation capabilities.</p>
```

>   <p>In this case study, we talk with Yashar Hosseinpour, who offers a detailed look at how LocalStack has streamlined Neurolytics' dev&test loops. This includes the reduction in development time, cost savings, and the overall impact on their engineering processes. We explore the challenges they faced, the solutions they implemented, and the benefits they've reaped from integrating LocalStack into their AWS-based development workflows.</p>

Reviewer: ```suggestion
  <p>In this case study, we talk with Yashar Hosseinpour, who offers some detailed insights into how LocalStack has streamlined Neurolytics' dev&test loops. This includes a reduction in development time, cost savings, and the overall impact on their engineering processes. We explore the challenges they faced, the solutions they implemented, and the benefits they've reaped from integrating LocalStack into their AWS-based development workflows.</p>
```

</div>

## Challenge

> Neurolytics faced considerable challenges in its development cycle due to the complexities of integrating with various AWS services. Their platform required the orchestration of multiple AWS services, which was a significant hurdle, as each update or feature development necessitated direct deployment and testing on AWS, which led to lengthy and inefficient development cycles. Moreover, the need to maintain individual AWS accounts for each developer added a layer of operational cost and complexity.

Reviewer: ```suggestion
Neurolytics faced considerable challenges in its development cycle due to the complexities of integrating with various AWS services. Their platform required the orchestration of multiple AWS services, which turned out to be a significant hurdle, as each update or feature development necessitated direct deployment and testing on AWS, which led to lengthy and inefficient development cycles. Moreover, the need to maintain individual AWS accounts for each developer added a layer of operational cost and complexity.
```

> The process of real-time testing and debugging in the AWS environment posed another significant challenge: The inability to perform quick local tests for immediate feedback considerably slowed down the pace of the development process. Consequently, the development team found themselves in a constant loop of building, deploying, and waiting, which was not only time-consuming but also detracted them from an agile development process.

Reviewer: ```suggestion
The process of real-time testing and debugging in the AWS environment posed another significant challenge: The inability to perform quick local tests for immediate feedback considerably slowed down the pace of development. Consequently, the development team found themselves in a constant loop of building, deploying, and waiting, which was not only time-consuming but also detracted them from an agile development process.
```

> In response to these challenges, Neurolytics began exploring solutions that could replicate the AWS environment locally. The aim was to establish a rapid, efficient feedback loop, enabling developers to test and debug in a local setup that closely mirrored the cloud environment. It was in this context that LocalStack emerged as a potential solution, promising to transform its cloud development and testing strategy.

Reviewer: ```suggestion
In response to these challenges, Neurolytics began exploring solutions that could replicate the AWS environment locally. The aim was to establish a rapid, efficient feedback loop, enabling developers to test and debug in a local setup that closely mirrored the cloud environment. It was in this context that LocalStack emerged as a potential solution, promising to transform their cloud development and testing strategy.
```

## Solution

> In response to the challenges faced in their development process, Neurolytics turned to LocalStack. This shift was geared towards achieving faster development iterations, reducing operational costs, and enhancing the overall efficiency of their development process. The integration of LocalStack into Neurolytics’ development marked a turning point. It allowed the team to run multiple cloud services locally, maintaining parity with the real AWS cloud.

Reviewer: ```suggestion
In response to the challenges faced in their development process, Neurolytics turned to LocalStack. This shift was geared towards achieving faster development iterations, reducing operational costs, and enhancing the overall efficiency of their development process. The integration of LocalStack into Neurolytics’ development toolstack marked a turning point. It allowed the team to run multiple cloud services locally, maintaining parity with the real AWS cloud.
```

This shift not only streamlined their development cycle but also offered an environment for quicker debugging and feature validation. The benefits extended beyond just technical improvements — financially, it translated into considerable cost savings by eliminating the need for individual AWS accounts for each developer, and operationally, it simplified the management of their cloud resources.

<div class="quote-container mt-4">

  > _“By eliminating the need to deploy main critical services to individual AWS accounts for each developer, we've saved around 20x per month per developer. With LocalStack, we've successfully streamlined our development process, making it a game-changer for our small team working on multiple cloud services.”_
  <div class="quote-author">
    <p><a href="https://www.linkedin.com/in/yashar-hosseinpour/">Yashar Hosseinpour</a>,</p>
    <p>Lead Developer at <a href="https://neurolytics.ai/">Neurolytics</a></p>
  </div>
</div>

> Moreover, Neurolytics developed a custom Command Line Interface (CLI) tool in Golang, leveraging AWS and Docker SDK, to manage their local infrastructure more effectively. This tool enabled them to start up their LocalStack environment with a single command and manage each service individually, further enhancing their development process.

Reviewer: ```suggestion
Moreover, Neurolytics developed a custom Command Line Interface (CLI) tool in Golang, leveraging AWS and Docker SDK, to manage their local infrastructure more effectively. This tool enabled them to start up their LocalStack environment with a single command and manage each service individually, further enhancing their processes and developer experience (DevX).
```

> The CLI also included features for backing up and exporting the state of services, as well as importing functionality, allowing for easy sharing of the local environment state among team members. This level of control and flexibility was instrumental in achieving a more efficient and collaborative development environment.

Reviewer: ```suggestion
The CLI also included features for backing up and exporting the state of services, as well as importing functionality, allowing for easy sharing of the local environment state among team members. This level of control and flexibility was instrumental in achieving a more efficient and collaborative development environment, powered by LocalStack.
```
